packages/components/pie-toast-provider/test/component/pie-toast-provider.spec.ts

```diff
@@ -1,17 +1,147 @@
+    type ExtendedToastProps,
+} from '../../src/defs.ts';
+
+import { toastProvider } from '../helpers/page-object/selectors.ts';
 
 test.describe('PieToastProvider - Component tests', () => {
-    test('should render successfully', async ({ mount, page }) => {
+    let toastsQueue: ExtendedToastProps[] = [];
+
+    test.beforeEach(({ page }) => {
+        // Set up a listener for the queue update log
+        page.on('console', async (message) => {
+            if (message.text().includes('toast provider queue:')) {
+                toastsQueue = await message.args()[1].jsonValue();
+            }
+        });
+    });
+
+    test('should render successfully', async ({ page }) => {
         // Arrange
-        await mount(PieToastProvider);
+        const pieToastProviderPage = new BasePage(page, 'toast-provider--default');
+        await pieToastProviderPage.load();
 
         // Act
-        const toastProvider = page.locator(componentSelector);
+        const toastProviderComponent = page.locator(toastProvider.selectors.container.dataTestId);
 
         // Assert
-        expect(toastProvider).toBeVisible();
+        await expect(toastProviderComponent).toBeDefined();
+    });
+
+    test.describe('Priority Order Tests', () => {
+        test('should handle toast priority correctly', async ({ page }) => {
+            // Arrange
+            const pieToastProviderPage = new BasePage(page, 'toast-provider--default');
+            await pieToastProviderPage.load();
+
+            // Act
+            await page.evaluate(() => {
+                const toastProvider = document.querySelector('pie-toast-provider') as PieToastProvider;
+
+                toastProvider.createToast({
+                    message: 'Neutral toast (Priority 10)',
+                    variant: 'neutral',
+                });
+
+                toastProvider.createToast({
+                    message: 'Success toast with action (Priority 4)',
+                    variant: 'success',
+                    leadingAction: { text: 'Action' },
+                });
+
+                toastProvider.createToast({
+                    message: 'Info toast with action (Priority 5)',
+                    variant: 'info',
+                    leadingAction: { text: 'Action' },
+                });
+
+                toastProvider.createToast({
+                    message: 'Error toast (Priority 2)',
+                    variant: 'error',
+                });
+            });
+
+            // Assert
+            const queueVariants = toastsQueue.map((toast: ExtendedToastProps) => `${toast.variant}${toast.leadingAction ? '-actionable' : ''}`);
+            for (let i = 1; i < queueVariants.length; i++) {
+                const prevPriority = PRIORITY_ORDER[queueVariants[i - 1]];
+                const currPriority = PRIORITY_ORDER[queueVariants[i]];
+                expect(currPriority).toBeGreaterThanOrEqual(prevPriority); // Ensure the current has a higher priority
+            }
```

Comment:
I wonder if a cleaner way might be to `sort` by priority and check that the ordering is the same?